pypeit/scripts/ql_keck_mosfire.py

```diff
@@ -198,14 +269,50 @@ def main(args):
                                           merge_with=config_lines(args))
     science_path = os.path.join(parset['rdx']['redux_path'], parset['rdx']['scidir'])
 
-    # Calibration Master directory
-    if args.master_dir is None:
-        msgs.error("You need to set an Environmental variable MOSFIRE_MASTERS that points at the Master Calibs")
+    # Parse the files sort by MJD
+    files = np.array([os.path.join(args.full_rawpath, file) for file in args.files])
+    nfiles = len(files)
+    mjds = np.zeros(nfiles)
+    for ifile, file in enumerate(files):
+        hdr = fits.getheader(file, spectrograph.primary_hdrext)
+        try:
+           mjds[ifile] = hdr['MJD-OBS']
+        except:
+            msgs.warn('File {:} has not MJD in header'.format(file))
+            mjds[ifile] = 0
+    files = files[np.argsort(mjds)]
+
+    # We need the platescale
+    platescale = spectrograph.get_detector_par(None, 1)['platescale']
+    # Parse the offset information out of the headers. TODO in the future get this out of fitstable
+    dither_pattern, dither_id, offset_arcsec = parse_dither_pattern(files, spectrograph.primary_hdrext)
+    if len(np.unique(dither_pattern)) > 1:
+        msgs.error('Currently this script is supported only for a single type of dither pattern')
+    A_files = files[dither_id == 'A']
+    B_files = files[dither_id == 'B']
+    nA = len(A_files)
+    nB = len(B_files)
+
+    # Print out a report on the offsets
+    msg_string = msgs.newline()  +     '****************************************************'
+    msg_string += msgs.newline() +     ' Summary of offsets for dither pattern:   {:s}'.format(dither_pattern[0])
+    msg_string += msgs.newline() +     '****************************************************'
+    msg_string += msgs.newline() +     'filename     Position         arcsec    pixels    '
+    msg_string += msgs.newline() +     '----------------------------------------------------'
+    for iexp, file in enumerate(files):
+        msg_string += msgs.newline() + '    {:s}    {:s}   {:6.2f}    {:6.2f}'.format(
+            os.path.basename(file), dither_id[iexp], offset_arcsec[iexp], offset_arcsec[iexp]/platescale)
+    msg_string += msgs.newline() +     '****************************************************'
+    msgs.info(msg_string)
+
+    #offset_dith_pix = offset_dith_pix = offset_arcsec_A[0]/sciImg.detector.platescale
+
 
     # Define some hard wired master files here to be later parsed out of the directory
-    slit_masterframe_name = os.path.join(args.master_dir, 'MasterSlits_E_15_01.fits.gz')
-    tilts_masterframe_name = os.path.join(args.master_dir, 'MasterTilts_E_1_01.fits')
-    wvcalib_masterframe_name = os.path.join(args.master_dir, 'MasterWaveCalib_E_1_01.fits')
+    slit_masterframe_name = os.path.join(args.master_dir, 'MasterSlits_D_1_01.fits.gz')
```

Comment:
I'd put these up at the start of the module and 
you should query that they exist.

Reply:
Done.